### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yermilov-m**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,60 @@
...
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String report = readData(fromFileName);
+        writeToFile(report, toFileName);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:13]: You miss report creating operation.
image

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,62 @@
...
+        String[] lineValue;
+        String nextString;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
+            nextString = reader.readLine();
+            while (nextString != null) {
+                lineValue = nextString.split(",");
+                if (lineValue[0].equals("supply")) {
+                    supply += Integer.parseInt(lineValue[1]);
+                } else if (lineValue[0].equals("buy")) {
+                    buy += Integer.parseInt(lineValue[1]);
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/WorkWithFile.java:29]: move magic numbers and string literals to constant

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,62 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String report = createReport(readData(fromFileName));
+        writeToFile(report, toFileName);
+    }
+
+    private int[] readData(String fromFileName) {
```
Comment (andriy-shulepa) [inline src/main/java/core/basesyntax/WorkWithFile.java:16]: this method not only reads the file but also performs calculations for the report. It's a violation of SPR. this method should only read a file and return a list of strings

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 2 above)
Comment (yermilov-m) [reply src/main/java/core/basesyntax/WorkWithFile.java:29]: why didn`t you request these changes earlier? why every time i check my homework, everyone finds out a few new mistakes i did from the start?

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 3 above)
Comment (yermilov-m) [reply src/main/java/core/basesyntax/WorkWithFile.java:16]: What is SPR? Strategic Petroleum Reserve? Society for Psychotherapy Research? Surface Plasmon Resonance? Were we learned this yet? If yes, please send me a link to a topic, I`ll check, but if we weren't... Ok, send it anyways

Comment (yermilov-m) [reply src/main/java/core/basesyntax/WorkWithFile.java:16]: But also read file and calc data in different places is a good idea anyways.

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**